rasa/shared/core/domain.py

```diff
@@ -591,6 +592,15 @@ def __init__(
         action_names += overridden_form_actions
 
         self.responses = responses
+        # checks if responses contain conditions
```

Comment:
optional: how about extracting that to a separate method / function? It in my opinion has a different level of abstraction than the other code in the constructor (http://principles-wiki.net/principles:single_level_of_abstraction)

Reply:
Yup, I'm not very happy about this check in the constructor - the question is where would I make this call to the separate method? Or should I just extract to an external function that I then call in the constructor?

Reply:
I've opted finally for a function that gets called in the constructor which I'll push in the next few minutes.